### PR TITLE
WIP - do not merge - Audit designate

### PR DESCRIPTION
**Pull request by notque**
WIP for audit-designate mapping, and rabbitmq-notifications change.

### CODE REVIEW

File: openstack/designate/requirements.yaml [hunk 1]
```diff
@@ -2,9 +2,14 @@ dependencies:
   - name: mariadb 
     repository: file://../../common/mariadb
     version: 0.1.1
-  - name: memcached
-    repository: file://../../common/memcached
-    version: 0.0.1
   - name: rabbitmq
+    alias: rabbitmq_notifications
```
Comment (jobrs) [inline openstack/designate/requirements.yaml]: this renames the existing (RPC-)rabbitmq. not good. We need an additional rabbitmq for notifications.

File: openstack/designate/templates/api-deployment.yaml [hunk 2]
```diff
@@ -42,7 +42,7 @@ spec:
             - name: DEPENDENCY_JOBS
               value: "designate-migration"
             - name: DEPENDENCY_SERVICE
-              value: "designate-mariadb,{{ .Release.Name }}-rabbitmq"
+              value: "designate-mariadb,rabbitmq"
```
Comment (jobrs) [inline openstack/designate/templates/api-deployment.yaml]: why this change? I did not introduce that dependency in Nova, Neutron, Cinder because the API should not be blocked by audit-specific components. If the notifications-rabbit is not up in time, the middleware will buffer.

File: openstack/designate/templates/central-deployment.yaml [hunk 3]
```diff
@@ -44,7 +44,7 @@ spec:
             - name: DEPENDENCY_JOBS
               value: "designate-migration"
             - name: DEPENDENCY_SERVICE
-              value: "designate-mariadb,{{ .Release.Name }}-rabbitmq"
```
Comment (jobrs) [inline openstack/designate/templates/central-deployment.yaml]: same as above - plus that central does not have the middleware

File: openstack/designate/templates/etc/_designate_audit_map.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    singleton: true
+  # Default mappings
+  tlds:
+  tsigkeys:
+  blacklists:
+  quotas:
+    singleton: true
+
+  service_statuses:
+    service_name: 'name'
```
Comment (jobrs) [inline openstack/designate/templates/etc/_designate_audit_map.yaml]: why service_name? the service_name is always the same. Only the resource typeURI varies.

Comment (jobrs) [reply openstack/designate/templates/etc/_designate_audit_map.yaml]: do we have a resource "service_statuses" with multiple instances? the typeURI of the elements would be 'service_statuse' unless you override it.

File: openstack/designate/values.yaml [hunk 5]
```diff
@@ -92,17 +92,6 @@ memcached:
   metrics:
     enabled: true
 
-rabbitmq:
```
Comment (jobrs) [inline openstack/designate/values.yaml]: the rabbitmq is not ours. Please do not touch

File: openstack/designate/values.yaml [hunk 6]
```diff
@@ -187,3 +176,22 @@ logging:
...
+  enabled: true
+  # do by default not attach the request payloads of create/update calls to the event
+  record_payloads: false
+  metrics_enabled: true
+  # how many messages to buffer before dumping to log (when rabbit is down or too slow)
+  mem_queue_size: 1000
+
+rabbitmq_notifications:
+  name: designate
+  # overwrite auth in region secrets
```
Comment (jobrs) [inline openstack/designate/values.yaml]: the default user-names are defined by the rabbitmq chart's value file already. Let's keep them unchanged to avoid confusion and remove this section here.

File: openstack/designate/templates/etc-configmap.yaml [hunk 7]
```diff
@@ -25,3 +25,8 @@ data:
 {{ include (print .Template.BasePath "/etc/_designate-policy.json.tpl") . | indent 4 }}
   logging.conf: |
 {{ include "loggerIni" .Values.logging | indent 4 }}
+
+{{- if .Values.audit.enabled }}
+    designate_audit_map.yaml: |
+{{ include (print .Template.BasePath "/etc/_designate_audit_map.yaml") . | indent 4 }}
```
Comment (jobrs) [inline openstack/designate/templates/etc-configmap.yaml]: how is this file copied to /etc/designate?

File: openstack/designate/templates/etc/_api-paste.ini.tpl [hunk 8]
```diff
@@ -5,6 +5,10 @@ use = egg:Paste#urlmap
 /v2: osapi_dns_v2
 /admin: osapi_dns_admin
 
+{{- define "audit_pipe" -}}
+{{- if .Values.audit.enabled }} audit{{- end -}}
```
Comment (jobrs) [inline openstack/designate/templates/etc/_api-paste.ini.tpl]: where is this added to the pipeline?

**Comment by jobrs** [comment]
also missing: changes to the designate config. Just updated the README to be more specific: https://github.com/sapcc/openstack-audit-middleware/blob/master/README.md#oslo-messaging

File: openstack/designate/requirements.yaml (hunk 1 above)
Comment (notque) [reply openstack/designate/requirements.yaml]: fixed requirements with your help.

File: openstack/designate/templates/api-deployment.yaml (hunk 2 above)
Comment (notque) [reply openstack/designate/templates/api-deployment.yaml]: It already had the requirement, I renamed it when I was renaming the rabbit. Removed the change.

File: openstack/designate/templates/central-deployment.yaml (hunk 3 above)
Comment (notque) [reply openstack/designate/templates/central-deployment.yaml]: removed the change.

File: openstack/designate/templates/etc/_designate_audit_map.yaml (hunk 4 above)
Comment (notque) [reply openstack/designate/templates/etc/_designate_audit_map.yaml]: /v2/service_status is a singleton that returns the health state of services. perhaps it shouldn't be included?

File: openstack/designate/values.yaml (hunk 5 above)
Comment (notque) [reply openstack/designate/values.yaml]: reverting

File: openstack/designate/values.yaml (hunk 6 above)
Comment (notque) [reply openstack/designate/values.yaml]: done

**Comment by notque** [comment]
designate.conf is setup, uses util include, and util added to requirements. we're fine there.

File: openstack/designate/templates/etc-configmap.yaml (hunk 7 above)
Comment (notque) [reply openstack/designate/templates/etc-configmap.yaml]: added copy now, without the template audit-enabled call, as the file is not a template right now.

**Comment by notque** [comment]
Just an update, it's working in staging, and I've sent the branch/pr to David to get a review. Waiting until that's sorted.
